rfcs/0000-unified-configuration.ftd

```diff
@@ -0,0 +1,312 @@
+-- import: fastn.com/rfcs/lib
+-- import: fastn-community.github.io/bling/note
+
+-- lib.rfc: RFC: Unified Configuration
+id: unified-config
+status: proposal
+
+A configuration system streamlining complex settings for processors, fastn
+packages. It introduces:
+
+- Special Configuration types in the `fastn` namespace.
+- A virtual module for configuration access in userland.
+- Package level configuration.
+
+-- lib.motivation:
+
+Currently, `fastn` supports configuration through [environment
+variables](/env/), cli arguments, and in some way, through the `FASTN.ftd`
+file. Environment variables are good to change the behaviour of the cli or to
+pass secret information but it's not ideal for defining complex configuration
+for a framework like `fastn` which provides various configurable features like
+[processors](/processor/-/backend/), oauth authentication and, apps.
+
+This RFC proposes a way of dealing with configuration that is centralized and
+leverages the existing ability of the `fastn` cli to read environment
+variables.
+
+-- end: lib.motivation
+
+
+-- lib.detailed-design:
+
+-- ds.h2: Configuration types in `fastn.*`
+
+This contains all special configuration types defined by the fastn framework.
+
+Any configuration required by the framework will be present as a type under the
+`fastn` namespace.
+
+For example, `db-conn` that represents database connection information is a type
+under `fastn` which can be used to create variables like:
+
+-- ds.code: variable in `FASTN.ftd` of `my-package`
+lang: ftd
+
+\-- import: fastn
+
+\-- fastn.package: my-package
+
+\-- fastn.db-conn my-db:
+db_host: <some_ip>
```

Comment:
We prefer `db-host` (kebab-case, not snake_case).

Reply:
Also for db we should allow "connections string/uri" as it incapsulates more than what you have written, eg https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING, and that's what we get from Heroku (https://devcenter.heroku.com/articles/heroku-postgresql#designating-a-primary-database), user is just going to pass the entire DATABASE_URL to us, and not part by part like you have shown in happy case.